### PR TITLE
adding arraybinding for nodejs

### PR DESCRIPTION
**Pull request by SimbaGithub**

### CODE REVIEW

File: lib/connection/bind_uploader.js [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+	
+		var dataRows = new Array();
+		for(var i=0; i<bindings.length; i++)
+		{
+			var strbuffer = "";
+			for(var j=0; j< bindings[i].length; j++)
+			{
+				if (j>0)
+					strbuffer += ',';
+				var value = bindings[i][j];
```
Comment (sfc-gh-hchaturvedi) [inline lib/connection/bind_uploader.js:63]: Isn't there possibility of special characters in here that might need to be escaped?

File: lib/connection/bind_uploader.js [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+	
+		var startIndex = 0;
+		var rowNum = 0;
+		var curBytes = 0;
+		var fileCount = 0;
+		
+	
+		while (rowNum < dataRows.length)
+		{
+			Logger.getInstance().debug('rowNum='+rowNum+', dataRows='+dataRows.length);
```
Comment (sfc-gh-hchaturvedi) [inline lib/connection/bind_uploader.js:78]: Can this entire loop not be done as part of the previous for loop where we're creating the CSV? Can the strbuffer created in the previous loop be ingested into data directly which is being done below?

File: lib/connection/bind_uploader.js (hunk 1 above)
Comment (SimbaGithub) [reply lib/connection/bind_uploader.js:63]: I tested with " and ', which work fine. User should escape it during input.

Comment (sfc-gh-hchaturvedi) [reply lib/connection/bind_uploader.js:63]: JDBC Driver escapes it. It would be great if we can remain consistent.

Comment (SimbaGithub) [reply lib/connection/bind_uploader.js:63]: nodejs took bind as an array, so it should handle the escape when user input to the array.

**Review by sfc-gh-hchaturvedi — APPROVED** [review]
LGTM. We can let the release move forward but please add a test for special characters.

File: lib/connection/bind_uploader.js (hunk 1 above)
Comment (SimbaGithub) [reply lib/connection/bind_uploader.js]: added cvs escape

File: test/integration/testArrayBind.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+		arrBind.push(['string'+i, i, "2020-05-11", "12:35:41.3333333", "2022-04-01 23:59:59", "2022-07-08 12:05:30.9999999"]);
+	}
+	
+	var insertStatement = connection.execute({
+		sqlText: insertWithQmark,
+		binds: arrBind,
+		complete: function (err, stmt) {
+			testUtil.checkError(err);
+			assert.strictEqual(stmt.getNumUpdatedRows(), count);
+			done();
```
Comment (sfc-gh-igarish) [inline test/integration/testArrayBind.js:52]: Can you try the test with removing this "done()"? This must be called in after() also.

**Review by SimbaGithub — COMMENTED** [review]
merge master to snow-373

**Comment by SimbaGithub** [comment]
For this PR, The user will not see any different, but it will increase the performance when they do bulk insert, the driver will convert that bulk insert data into different files and upload to the server, instead of sending everything at once.

**Pull request merged**